content/en/blog/_posts/2021-08-20-seccomp-default.md

```diff
@@ -0,0 +1,260 @@
+are unlikely or unsafe to be used in a containerized application.
+
+[crio]: https://github.com/cri-o/cri-o/blob/fe30d62/vendor/github.com/containers/common/pkg/seccomp/default_linux.go#L45
+[ctrd]: https://github.com/containerd/containerd/blob/e1445df/contrib/seccomp/seccomp_default.go#L51
+
+### Enabling the feature
+
+Two kubelet configuration changes have to be made to enable the feature:
+
+1. set the `SeccompDefault=true` via the command line (`--feature-gates`) or the
+   [kubelet configuration][kubelet] file
+2. enable the feature by adding the `--seccomp-default` command line flag or via
+   the [kubelet configuration][kubelet] file (`seccompDefault: true`)
+
+[kubelet]: /docs/tasks/administer-cluster/kubelet-config-file
+
+The kubelet will error on startup if only one of the above steps have been done.
+
+### Trying it out
+
+If the feature is enabled on a node, then you can create a new workload like
+this:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-pod
+spec:
+  containers:
+    - name: test-container
+      image: nginx:1.21
+```
+
+Now it is possible to inspect the used seccomp profile by using
+[`crictl`][crictl] while investigating the containers [runtime
+specification][rspec]:
+
+[crictl]: https://github.com/kubernetes-sigs/cri-tools
+[rspec]: https://github.com/opencontainers/runtime-spec/blob/0c021c1/config-linux.md#seccomp
+
+```bash
+CONTAINER_ID=$(sudo crictl ps -q --name=test-container)
+sudo crictl inspect $CONTAINER_ID | jq .info.runtimeSpec.linux.seccomp
+```
+
+```yaml
+{
+  "defaultAction": "SCMP_ACT_ERRNO",
+  "architectures": ["SCMP_ARCH_X86_64", "SCMP_ARCH_X86", "SCMP_ARCH_X32"],
+  "syscalls": [{ "names": [
+            "_llseek",
+            "_newselect",
+            "accept",
+            …
+            "write",
+            "writev",
+          ], "action": "SCMP_ACT_ALLOW" }, …],
+}
+```
+
+You can see that the lower level container runtime ([CRI-O][crio-home] and
+[runc][runc] in our case), successfully applied the default seccomp profile.
+This profile denies all syscalls per default, while allowing commonly used ones
+like [`accept`][accept] or [`write`][write].
+
+[crio-home]: https://github.com/cri-o/cri-o
+[runc]: https://github.com/opencontainers/runc
+[accept]: https://man7.org/linux/man-pages/man2/accept.2.html
+[write]: https://man7.org/linux/man-pages/man2/write.2.html
+
+Please note that the feature will not influence any Kubernetes API for now.
+Therefore, it is not possible to retrieve the used seccomp profile via `kubectl`
+`get` or `describe` if the [`SeccompProfile`][api] field is unset within the
+`SecurityContext`.
+
+The feature also works when using multiple containers within a pod, for example
+if you create a pod like this:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test-pod
+spec:
+  containers:
+    - name: test-container-nginx
+      image: nginx:1.21
+      securityContext:
+        seccompProfile:
```

Comment:
One thing that might be worth clarifying is how this field works for privileged containers:

Does setting `privileged: true` automatically set the `seccompProfile` to `Unconfined` or does it have to be set in addition to `privileged: true`

If it is the latter, many pod yamls that set `privileged: true` might need an update to add the `seccompProfile` field to `Unconfined`, when running on nodes with this feature enabled. 

If this is already mentioned somewhere in other docs, it would be good to link that here somewhere in the blog post.

Reply:
Hey, the behavior does not change from the current one: Privileged containers will always run as unconfined.

Reply:
Added a follow-up to the docs: https://github.com/kubernetes/website/pull/29489

Reply:
Thank you for making this change